tests/test_connection.py

```diff
@@ -271,6 +272,36 @@ def mock_request(*args, **kwargs):
         with pytest.raises(RequestException, match=r"Failed to resolve hostname 'test.xanadu.ai'"):
             connection.request("GET", "/healthz")
 
+    @responses.activate
+    def test_request_headers(self, connection):
+        """Test that request() passes the headers attribute to _request if the headers argument
+        is not provided."""
+
+        responses.add(
+            url=connection.url("path"),
+            method="POST",
+            status=200,
+            match=(matchers.header_matcher(connection.headers),),
+        )
+
+        assert connection.request(method="POST", path="path").status_code == 200
+
+    @responses.activate
+    @pytest.mark.parametrize("extra_headers", [{"X-Test": "data"}, {}])
+    def test_request_extra_headers(self, connection, extra_headers):
+        """Tests that request() passes the combined headers from the headers attribute
+        and the headers argument to the _request method."""
+        responses.add(
+            url=connection.url("path"),
+            method="POST",
+            status=200,
+            match=(matchers.header_matcher({**connection.headers, **extra_headers}),),
+        )
+
+        assert (
+            connection.request(method="POST", path="path", headers=extra_headers).status_code == 200
+        )
```

Comment:
Similar to the above,
```suggestion
        connection.request(method="POST", path="path", headers=extra_headers)
```